Source/WTF/wtf/HashTable.h

```diff
@@ -631,44 +631,35 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(HashTable);
     {
     }
 
-#if !ASSERT_ENABLED
-
-    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits>
+    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits, ShouldValidateKey validateKey>
     template<typename HashTranslator, typename T>
-    inline void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits>::checkKey(const T&)
+    void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits, validateKey>::checkKey(const T& key)
     {
-    }
-
-#else // ASSERT_ENABLED
+        if constexpr (validateKey == ShouldValidateKey::No)
```

Comment:
It's a little risky to not do this at call sites. If the function doesn't get inlined, then we'll pay a new function call in unsafe HashMaps, which could hurt perf.